content/articles/2019-09-tidyr-1-0-0.Rmarkdown

```diff
@@ -0,0 +1,316 @@
+    ```
+    
+    This is a designed experiment with nine replicates for every combination of
+    `wool` (`A` and `B`) and `tension` (`L`, `M`, `H`). If we attempt to pivot 
+    the levels of `wool` into the columns we get a warning and the output
+    contains list-columns:
+    
+    ```{r}
+    warpbreaks %>% pivot_wider(names_from = wool, values_from = breaks)
+    ```
+
+    You can now summarise the duplicates with the `values_fn` argument:
+    
+    ```{r}
+    warpbreaks %>% 
+      pivot_wider(
+        names_from = wool, 
+        values_from = breaks,
+        values_fn = list(breaks = mean)
+      )
+    ```
+
+Learn the full details and see many more examples in [`vignette("pivot")`](http://tidyr.tidyverse.org/dev/articles/pivot.html).
+
+## Rectangling
+
+Rectangling is the art and craft of taking a deeply nested list (often sourced from wild caught JSON or XML) and taming it into a tidy data set of rows and columns. tidyr 1.0.0 provides four new functions to aid rectangling:
+
+* `unnest_longer()` takes each element of a list-column and makes a new row.
+* `unnest_wider()` takes each element of a list-column and makes a new column.
+* `unnest_auto()` uses some heuristics to guess whether you want 
+  `unnest_longer()` or `unnest_wider()`.
+* `hoist()` is similar to `unnest_wider()` but only plucks out selected
+  components, and can reach down multiple levels.
+
+To see them in action, take this small sample of `repurrrsive::got_chars`. It contains data about three characters from the Game of Thrones:
+
+```{r}
+characters <- list(
+  list(
+    name = "Theon Greyjoy",
+    aliases = c("Prince of Fools", "Theon Turncloak", "Theon Kinslayer"),
+    alive = TRUE
+  ),
+  list(
+    name = "Tyrion Lannister",
+    aliases = c("The Imp", "Halfman", "Giant of Lannister"),
+    alive = TRUE
+  ),
+  list(
+    name = "Arys Oakheart",
+    alive = FALSE
+  )
+)
+```
+
+To work with the new tidyr rectangling tools, we first put the list into a data frame, creating a list-column:
+
+```{r}
+got <- tibble(character = characters)
+got
+```
+
+We can then use `unnest_wider()` to make each element of that list into a column:
+
+```{r}
+got %>% 
+  unnest_wider(character)
+```
+
+Followed by `unnest_longer()` to turn each alias into its own row:
+
+```{r}
+got %>% 
+  unnest_wider(character) %>% 
+  unnest_longer(aliases)
+```
+
+Even more conveniently, you can use `unnest_auto()` to guess which direction a list column should be unnested in. Here it yields the same results as above, and the messages tell you why:
+
+```{r, results = FALSE}
+got %>% 
+  unnest_auto(character) %>% 
+  unnest_auto(aliases)
+```
+
+Alternatively, you can use `hoist()` to reach deeply into a data structure and put out just the pieces you need:
+
+```{r}
+got %>% hoist(character, 
+  name = "name",
+  alias = list("aliases", 1),
+  alive = "alive"
+)
+```
+
+This replaces our previous recommendation to use `dplyr::mutate()` and `purrr::map()` together:
```

Comment:
Possible re-framing:

> This syntax provides a more approachable alternative to using `purrr::map()` inside `dplyr::mutate()`:

Where I'm coming from: convey both are correct and supported, but show some solidarity with those who find "map inside mutate" rather mind-bendy and verbose.

Reply:
I'm with Jenny here. I think the word "replaces" sounds an alarm so loud it drowns out "recommendation."